### PR TITLE
feat: :sparkles: authentication flow

### LINKED ISSUE #3: [FEAT] :sparkles: implement deserializeUser middleware
### Description

this middleware will get user from session or cookie and then save it to `res.local.user`

### Anything else?

file:- `/src/middleware/deserializeUser.middleware.ts`

reference from another project:- https://github.com/shriproperty/server/blob/dev/src/middlewares/deserializeUser.middleware.ts

### LINKED ISSUE #16: [REFACTOR] :recycle: JWT auth
### Description

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by aayushchugh**
### Description

Implemented authentication workflow, new routes created for
- Verify user
- login
- forgot password
- reset password
- logout
- get current user
- refresh access token

### Checklist:

- [x] This pull request follow our contribution guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (postman and swagger)
- [ ] I have written tests for the code
- [ ] I have updated .env.example file for new .env variables

### What is current behavior?

Closes #16

### CODE REVIEW

File: src/controllers/auth.controller.ts [hunk 1]
```diff
@@ -56,3 +73,299 @@ export async function signupHandler(
...
 }
+
+export async function verifyUserHandler(
+	req: Request<VerifyUserSchema["params"]>,
+	res: Response
+) {
+	const { email, verificationCode } = req.params;
+
+	try {
+		// QUESTION: should we use username to find the user or email?
```
Comment (is-it-ayush) [inline src/controllers/auth.controller.ts:84]: In the verification, Keep it Email! For the dash, keep it both "Username and Email."

Comment (aayushchugh) [reply src/controllers/auth.controller.ts:84]: Ok

**Comment by aayushchugh** [comment]
@is-it-ayush please approve this pull request also so that it can be merged

**Review by is-it-ayush — APPROVED** [review]
Flawless code with Good REST practices. Great Work Ayush!

**Review by BitOfAByte — APPROVED** [review]
I like it good job

**Pull request merged**